postgres-appliance/postgres_ha.py

```diff
@@ -0,0 +1,286 @@
+        config = {'etcd': defaults['etcd']}
+        config['etcd']['host'] = placeholders['ETCD_HOST']
+    elif 'ETCD_DISCOVERY_DOMAIN' in placeholders:
+        config = {'etcd': defaults['etcd']}
+        config['etcd']['discovery_srv'] = placeholders['ETCD_DISCOVERY_DOMAIN']
+    else:
+        pass  # Configuration can also be specified using PATRONI_CONFIGURATION
+
+    return config
+
+
+def write_wale_command_environment(placeholders):
+    placeholders['instance_data']['ip'] = get_instance_meta_data('local-ipv4')
+    az = get_instance_meta_data('placement/availability-zone')
+
+    if not os.path.exists(placeholders['WALE_ENV_DIR']):
+        os.makedirs(placeholders['WALE_ENV_DIR'])
+
+    with open(os.path.join(placeholders['WALE_ENV_DIR'], 'WALE_S3_PREFIX'), 'w') as f:
+        f.write('s3://{WAL_S3_BUCKET}/spilo/{SCOPE}/wal/'.format(**placeholders))
+
+    with open(os.path.join(placeholders['WALE_ENV_DIR'], 'WALE_S3_ENDPOINT'), 'w') as f:
+        f.write('https+path://s3-{}.amazonaws.com:443'.format(az[:-1]))
+
+
+def write_crontab(placeholders, path):
+    lines = ['PATH={}'.format(path)]
+    lines += ['{BACKUP_SCHEDULE} /postgres_backup.sh "{WALE_ENV_DIR}" "{PGDATA}"'.format(**placeholders)]
+    lines += yaml.load(placeholders['CRONTAB'])
+    lines += ['']  # EOF requires empty line for cron
+
+    c = subprocess.Popen(['crontab'], stdin=subprocess.PIPE)
+    c.communicate(input='\n'.join(lines).encode())
+
+
+def configure_patronictl(patroni_configfile, patronictl_configfile):
+    if not os.path.exists(os.path.dirname(patronictl_configfile)):
+        os.makedirs(os.path.dirname(patronictl_configfile))
+    if not os.path.exists(patronictl_configfile):
+        os.symlink(patroni_configfile, patronictl_configfile)
+
+
+def main():
+    debug = os.environ.get('DEBUG', '') in ['1', 'true', 'on', 'ON']
+    logging.basicConfig(format='%(asctime)s - bootstrapping - %(levelname)s - %(message)s', level=('DEBUG'
+                         if debug else 'INFO'))
+    if debug:
+        os.execlpe('bash', 'bash', dict(os.environ))
+
+    placeholders = get_placeholders()
+
+    write_certificates(placeholders)
+
+    config = yaml.load(pystache_render(TEMPLATE, placeholders))
+    config.update(get_dcs_config(config, placeholders))
+
+    config = deep_update(yaml.load(os.environ.get('PATRONI_CONFIGURATION') or '{}'), config)
+
+    # Patroni configuration
+    patroni_configfile = os.path.join(placeholders['PGHOME'], 'postgres.yml')
+    write_configuration(config, patroni_configfile)
+
+    configure_patronictl(patroni_configfile, os.path.join(placeholders['PGHOME'], '.config', 'patroni',
+                         'patronictl.yaml'))
+
+    # WAL-E
+    write_wale_command_environment(placeholders)
+
+    # # We run cron, to schedule recurring tasks on the node
+    logging.info('Starting up cron')
+    subprocess.call(['/usr/bin/sudo', '/usr/sbin/cron'])
+    write_crontab(placeholders, os.environ.get('PATH'))
```

Comment:
Would it make more sense to first write the crontab, and only afterwards launch the cron daemon?